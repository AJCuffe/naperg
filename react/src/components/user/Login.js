import React, { Component } from 'react'
import { AUTH_TOKEN } from '../../constants/constants'
import { graphql, compose } from 'react-apollo'
import gql from 'graphql-tag'
import SnackBarCustom from './SnackBarCustom'
const queryString = require('query-string')

class Login extends Component {
  state = {
    stateLogin: 'login', // switch between Login and SignUp
    email: '',
    password: '',
    name: '',
    messageSnackBar: '',
    openSnackBar: false,
    resetPasswordToken: ''
  }
  componentDidMount() {
    let resetPasswordToken = queryString.parse(this.props.location.search).resetPasswordToken
    if(resetPasswordToken) {
      // console.log(resetPasswordToken)
      this.setState({
        stateLogin: 'resetPassword',
        resetPasswordToken: resetPasswordToken
      })
    }
  }

  render() {
    return (
      <div>
        <h4 className='mv3'>
          {this.state.stateLogin === 'login' && 'Login'}
          {this.state.stateLogin === 'signup' && 'Sign Up'}
          {this.state.stateLogin === 'forget' && 'Forget Password'}
        </h4>
        <div className='flex flex-column'>
          {this.state.stateLogin === 'signup' && (
            <input
              value={this.state.name}
              onChange={e => this.setState({ name: e.target.value })}
              type='text'
              placeholder='Your name'
            />
          )}
          {(
            this.state.stateLogin === 'login' ||
            this.state.stateLogin === 'signup' ||
            this.state.stateLogin === 'forget'
          ) && (
          <input
            value={this.state.email}
            onChange={e => this.setState({ email: e.target.value })}
            type='text'
            placeholder='Your email address'
          />
        )}
        {(
          this.state.stateLogin === 'signup' ||
          this.state.stateLogin === 'login' ||
          this.state.stateLogin === 'resetPassword'
        ) && (
          <input
            value={this.state.password}
            onChange={e => this.setState({ password: e.target.value })}
            type='password'
            placeholder='Choose a safe password'
          />
      )}
        </div>
        <div className='flex mt3'>
          <div className='pointer f6 link dim br1 ba ph3 pv2 fr mb2 dib black' onClick={() => this._confirm()}>
            Ok
          </div>
          <div
            className='pointer f6 link dim br1 ba ph3 pv2 fr mb2 dib black'
            onClick={() => this.setState({ stateLogin: 'login', openSnackBar: false })}
          >Login
          </div>
          <div
            className='pointer f6 link dim br1 ba ph3 pv2 fr mb2 dib black'
            onClick={() => this.setState({ stateLogin: 'signup', openSnackBar: false })}
          >signup
          </div>
          <div
            className='pointer f6 link dim br1 ba ph3 pv2 fr mb2 dib black'
            onClick={() => this.setState({ stateLogin: 'forget', openSnackBar: false })}
          >Forget Password
          </div>
        </div>
        <SnackBarCustom
          openSnackBar={this.state.openSnackBar}
          messageSnackBar={this.state.messageSnackBar}/>
      </div>
    )
  }

  _confirm = async () => {
    const { name, email, password, resetPasswordToken } = this.state
    if (this.state.stateLogin === 'login') {
      const result = await this.props.loginMutation({
        variables: {
          email,
          password,
        },
      })
      const { token, user } = result.data.login
      this._saveUserData(token, user)
      this.props.history.push(`/`)
    }
    if (this.state.stateLogin === 'forget') {
      const result = await this.props.forgetPasswordMutation({
        variables: {
          email
        },
      })
      let messageSnackBar = `A mail has been sent with a
        link available until
        ${new Date(result.data.forgetPassword.resetPasswordExpires).toLocaleString()}`
      this.setState({
        messageSnackBar: messageSnackBar,
        openSnackBar: true,
        stateLogin: 'login'

      })
    }
    if (this.state.stateLogin === 'resetPassword') {
      const result = await this.props.resetPasswordMutation({
        variables: {
          resetPasswordToken,
          password
        },
      })

      const { token, user } = result.data.resetPassword
      this._saveUserData(token, user)
      this.props.history.push(`/`)

    }
    if (this.state.stateLogin === 'signup') {
      const result = await this.props.signupMutation({
        variables: {
          name,
          email,
          password,
        },
      })
      const { token, user } = result.data.signup
      this._saveUserData(token, user)
      this.props.history.push(`/`)
    }
  }

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token)
    localStorage.setItem('userToken', JSON.stringify(user))
  }
}

const SIGNUP_MUTATION = gql`
  mutation SignupMutation($email: String!, $password: String!, $name: String!) {
    signup(email: $email, password: $password, name: $name) {
      token
      user {
        name
        id
      }
    }
  }
`

const LOGIN_MUTATION = gql`
  mutation LoginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        name
        id
      }
    }
  }
`
const FORGET_PASSWORD_MUTATION = gql`
  mutation ForgetPasswordMutation($email: String!) {
    forgetPassword(email: $email) {
      name
      id
      resetPasswordExpires
    }
  }
`
const RESET_PASSWORD_MUTATION = gql`
  mutation ResetPasswordMutation($password: String!, $resetPasswordToken: String!) {
    resetPassword(password: $password, resetPasswordToken: $resetPasswordToken) {
      token
      user {
        name
        id
      }
    }
  }
`

export default compose(
  graphql(SIGNUP_MUTATION, { name: 'signupMutation' }),
  graphql(LOGIN_MUTATION, { name: 'loginMutation' }),
  graphql(FORGET_PASSWORD_MUTATION, { name: 'forgetPasswordMutation' }),
  graphql(RESET_PASSWORD_MUTATION, { name: 'resetPasswordMutation' }),
)(Login)
