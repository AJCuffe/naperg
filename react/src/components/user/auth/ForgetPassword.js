import React, { Component } from 'react'
import { graphql, compose } from 'react-apollo'
import gql from 'graphql-tag'
import SnackBarCustom from '../SnackBarCustom'
import Paper from 'material-ui/Paper'
import Button from 'material-ui/Button'


class ForgetPassword extends Component {
  state = {
    email: '',
    password: '',
    name: '',
    messageSnackBar: '',
    openSnackBar: false,
    resetPasswordToken: '',
    validateEmailToken: '',
  }


  render() {
    return (
      <div className='paperOut'>
        <Paper className='paperIn'>
        <h4 className='mv3'>
          Forget Password
        </h4>
        <div className='flex flex-column'>

          <input
            value={this.state.email}
            onChange={e => this.setState({ email: e.target.value })}
            type='text'
            placeholder='Your email address'
          />

        </div>
        <div className='flex mt3'>
          <Button variant='raised' onClick={() => this._confirm()}>
            Ok
          </Button>
        </div>
        <SnackBarCustom
          openSnackBar={this.state.openSnackBar}
          messageSnackBar={this.state.messageSnackBar}/>
      </Paper>
      </div>
    )
  }

  _confirm = async () => {
    const { email } = this.state
      let messageSnackBar
      await this.props.forgetPasswordMutation({
        variables: {
          email
        },
      })
      .then((result) => {
        messageSnackBar = `A mail has been sent with a link available until
        ${new Date(result.data.forgetPassword.resetPasswordExpires).toLocaleString()}`
      })
      .catch((e) => { messageSnackBar = e.graphQLErrors[0].message })

      this.setState({
        messageSnackBar: messageSnackBar,
        openSnackBar: true,
      })
  }

}



const FORGET_PASSWORD_MUTATION = gql`
  mutation ForgetPasswordMutation($email: String!) {
    forgetPassword(email: $email) {
      name
      id
      resetPasswordExpires
    }
  }
`

export default compose(

  graphql(FORGET_PASSWORD_MUTATION, { name: 'forgetPasswordMutation' }),

)(ForgetPassword)
