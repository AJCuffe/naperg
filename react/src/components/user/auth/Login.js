import React, { Component } from 'react'
import { AUTH_TOKEN } from '../../../constants/constants'
import { graphql, compose } from 'react-apollo'
import gql from 'graphql-tag'
import SnackBarCustom from '../SnackBarCustom'
import Paper from 'material-ui/Paper'
import Button from 'material-ui/Button'

const queryString = require('query-string')

class Login extends Component {
  state = {
    stateLogin: 'login',
    email: '',
    password: '',
    name: '',
    messageSnackBar: '',
    openSnackBar: false,
    resetPasswordToken: '',
    validateEmailToken: '',
  }

  componentDidMount() {
    let resetPasswordToken = queryString.parse(this.props.location.search).resetPasswordToken
    if(resetPasswordToken) {
      this.setState({
        stateLogin: 'resetPassword',
        resetPasswordToken: resetPasswordToken
      })
    }
    let validateEmailToken = queryString.parse(this.props.location.search).validateEmailToken
    if(validateEmailToken) {
      this.validateEmailMutation(validateEmailToken)
    }
  }

  render() {
    return (
      <div className='paperOut'>
        <Paper className='paperIn'>
        <h4 className='mv3'>
          {this.state.stateLogin === 'login' && 'Login'}
          {this.state.stateLogin === 'forget' && 'Forget Password'}
        </h4>
        <div className='flex flex-column'>
          {(
            this.state.stateLogin === 'login' ||
            this.state.stateLogin === 'forget'
          ) && (
          <input
            value={this.state.email}
            onChange={e => this.setState({ email: e.target.value })}
            type='text'
            placeholder='Your email address'
          />
        )}
        {(
          this.state.stateLogin === 'login' ||
          this.state.stateLogin === 'resetPassword'
        ) && (
          <input
            value={this.state.password}
            onChange={e => this.setState({ password: e.target.value })}
            type='password'
            placeholder='Choose a safe password'
          />
      )}
        </div>
        <div className='flex mt3'>
          <Button variant='raised' onClick={() => this._confirm()}>
            Ok
          </Button>
          <Button variant='flat'
            onClick={() => this.props.history.push('/signup')}
          >signup
          </Button>
          <Button variant='flat'
            onClick={() => this.setState({ stateLogin: 'forget', openSnackBar: false })}
          >Forget Password
          </Button>
        </div>
        <SnackBarCustom
          openSnackBar={this.state.openSnackBar}
          messageSnackBar={this.state.messageSnackBar}/>
      </Paper>
      </div>
    )
  }
  validateEmailMutation = async (validateEmailToken) => {
    let messageSnackBar
    await this.props.validateEmailMutation({
      variables: {
        validateEmailToken
      },
    })
    .then((result) => { messageSnackBar = `${result.data.validateEmail.email} is now validated.` })
    .catch((e) => { messageSnackBar = e.graphQLErrors[0].message })

    this.setState({
      messageSnackBar: messageSnackBar,
      openSnackBar: true,
      stateLogin: 'login'
    })
  }
  _confirm = async () => {
    const { name, email, password, resetPasswordToken } = this.state
    if (this.state.stateLogin === 'login') {
      await this.props.loginMutation({
        variables: {
          email,
          password,
        },
      })
      .then((result) => {
        const { token, user } = result.data.login
        this._saveUserData(token, user)
        this.props.history.push(`/`)
      })
      .catch((e) => {
        let messageSnackBar = e.graphQLErrors[0].message
        this.setState({
          messageSnackBar: messageSnackBar,
          openSnackBar: true,
        })
      })
    }
    if (this.state.stateLogin === 'forget') {
      let messageSnackBar
      await this.props.forgetPasswordMutation({
        variables: {
          email
        },
      })
      .then((result) => { messageSnackBar = `A mail has been sent with a link available until
        ${new Date(result.data.forgetPassword.resetPasswordExpires).toLocaleString()}` })
      .catch((e) => { messageSnackBar = e.graphQLErrors[0].message })

      this.setState({
        messageSnackBar: messageSnackBar,
        openSnackBar: true,
        stateLogin: 'login'
      })
    }
    if (this.state.stateLogin === 'resetPassword') {
      const result = await this.props.resetPasswordMutation({
        variables: {
          resetPasswordToken,
          password
        },
      })

      const { token, user } = result.data.resetPassword
      this._saveUserData(token, user)
      this.props.history.push(`/`)

    }
  }

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token)
    localStorage.setItem('userToken', JSON.stringify(user))
  }
}



const LOGIN_MUTATION = gql`
  mutation LoginMutation($email: String!, $password: String!) {
    login(email: $email, password: $password) {
      token
      user {
        name
        id
      }
    }
  }
`
const VALIDATE_EMAIL_TOKEN_MUTATION = gql`
  mutation ValidateEmailMutation($validateEmailToken: String!) {
    validateEmail(validateEmailToken: $validateEmailToken) {
      email
    }
  }
`
const FORGET_PASSWORD_MUTATION = gql`
  mutation ForgetPasswordMutation($email: String!) {
    forgetPassword(email: $email) {
      name
      id
      resetPasswordExpires
    }
  }
`
const RESET_PASSWORD_MUTATION = gql`
  mutation ResetPasswordMutation($password: String!, $resetPasswordToken: String!) {
    resetPassword(password: $password, resetPasswordToken: $resetPasswordToken) {
      token
      user {
        name
        id
      }
    }
  }
`

export default compose(
  graphql(LOGIN_MUTATION, { name: 'loginMutation' }),
  graphql(FORGET_PASSWORD_MUTATION, { name: 'forgetPasswordMutation' }),
  graphql(RESET_PASSWORD_MUTATION, { name: 'resetPasswordMutation' }),
  graphql(VALIDATE_EMAIL_TOKEN_MUTATION, { name: 'validateEmailMutation' }),
)(Login)
