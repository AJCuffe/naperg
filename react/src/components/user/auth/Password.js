import React, { Component } from 'react'
import Button from '@material-ui/core/Button'
import Input from '@material-ui/core/Input'
import InputAdornment from '@material-ui/core/InputAdornment'
import Icon from '@material-ui/core/Icon'
import InputLabel from '@material-ui/core/InputLabel'
import FormControl from '@material-ui/core/FormControl'
import FormHelperText from '@material-ui/core/FormHelperText'


export default class Password extends Component {
  state = {
    // email: '',
    // emailValidation: true,
    inputValidation2: true,
    isPasswordLongEnough: true,
    hasLowerCase: true,
    password: '',
    // name: '',
    // nameFile: '',
    // activeStep: 0,
    // maxStep: 3,
    passwordMinimumLength: 10
  }
  componentWillReceiveProps(newProps) {
    this.input2.focus()
  }
  onChange2(e){
    this.props.onChange2(e)
    this.setState({
      hasLowerCase: this.hasLowerCase(e.target.value),
      isPasswordLongEnough: this.isPasswordLongEnough(e.target.value)
    })
    if(this.state.hasLowerCase && this.state.isPasswordLongEnough) {
      this.setState({inputValidation2: true})
    } else {
      this.setState({inputValidation2: false})
    }
  }

  hasLowerCase(str) {
     if(str.toUpperCase() !== str) {
       return true
     }
     return false
  }

  isPasswordLongEnough(password) {
    if(password.length > this.state.passwordMinimumLength) {
      this.setState({isPasswordLongEnough: true})
      return true
    }
    this.setState({isPasswordLongEnough: false})
    return false
  }



  handleNext = () => {
    this.props.handleNext()
  };

  handleKey = (data) => {
    if(data.charCode === 13) { //keyPress enter
      this.handleNext()
    }
  }


  render() {
    return (

              <FormControl className={'wrapperAnimate ' + (this.props.activeStep ? 'focusField' : 'notFocusField')}>
                <InputLabel htmlFor='password'>Choose a safe password</InputLabel>
                <Input
                  id='password'
                  value={this.props.password}
                  error={!this.state.inputValidation2}
                  onChange={this.onChange2.bind(this)}
                  type='password'
                  inputRef={node => this.input2 = node}
                  onKeyPress={this.handleKey}
                  endAdornment={
                    <InputAdornment position='end'>
                      {this.props.activeStep && (
                        <Button onClick={this.handleNext} variant='fab' color='primary' mini>
                          <Icon>done</Icon>
                        </Button>
                      )}
                    </InputAdornment>
                  }
                />
              {!this.state.isPasswordLongEnough && (
                <FormHelperText>At least {this.state.passwordMinimumLength} characters long.</FormHelperText>
              )}
              {!this.state.hasLowerCase && (
                <FormHelperText>At least a lower case.</FormHelperText>
              )}
          </FormControl>

    )
  }
}
