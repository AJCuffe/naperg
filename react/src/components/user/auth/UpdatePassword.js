import React, { Component } from 'react'
import { AUTH_TOKEN } from '../../../constants/constants'
import { graphql, compose } from 'react-apollo'
import gql from 'graphql-tag'
import SnackBarCustom from '../../nav/SnackBarCustom'
import Paper from 'material-ui/Paper'
import Button from 'material-ui/Button'
import TextField from 'material-ui/TextField'

// const queryString = require('query-string')

class ChangePassword extends Component {
  state = {
    oldPassword: '',
    newPassword: '',
    newPassword2: '',
  }

  componentDidMount() {

  }

  render() {
    return (
      <div className='paperOut'>
        <Paper className='paperIn'>
        <h4 className='mv3'>
          Change Password
        </h4>
        <div className='flex flex-column'>

          <TextField
            value={this.state.oldPassword}
            onChange={e => this.setState({ oldPassword: e.target.value })}
            type='password'
            label='Your actual password'
          />
          <TextField
            value={this.state.newPassword}
            onChange={e => this.setState({ newPassword: e.target.value })}
            type='password'
            label='Choose a safe password'
          />
          <TextField
            value={this.state.newPassword2}
            onChange={e => this.setState({ newPassword2: e.target.value })}
            type='password'
            label='Retype your safe password'
          />

        </div>
        <div className='flex mt3'>
          <Button variant='raised' onClick={() => this._confirm()}>
            Ok
          </Button>

        </div>
        <SnackBarCustom ref={instance => { this.child = instance }}/>
      </Paper>
      </div>
    )
  }

  _confirm = async () => {
    if(!this.state.newPassword || !this.state.newPassword2 || !this.state.oldPassword) {
      this.child._openSnackBar('Password cannot be null')
      return
    }
    if(this.state.newPassword !== this.state.newPassword2) {
      this.child._openSnackBar('Error: Passwords are differents')
      return
    }
    let messageSnackBar
    const { oldPassword, newPassword } = this.state
    await this.props.updatePasswordMutation({
      variables: {
        oldPassword,
        newPassword
      },
    })
    .then((result) => {
      messageSnackBar = `Your password has been changed successfully!`
      this.setState({
        oldPassword: '',
        newPassword: '',
        newPassword2: '',
      })

    })
    .catch((e) => {
      // console.log(e)
      messageSnackBar = e.graphQLErrors[0].message
    })
    this.child._openSnackBar(messageSnackBar)
  }

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token)
    localStorage.setItem('userToken', JSON.stringify(user))
  }
}



const UPDATE_PASSWORD_MUTATION = gql`
  mutation UpdatePasswordMutation($oldPassword: String!, $newPassword: String!) {
    updatePassword(oldPassword: $oldPassword, newPassword: $newPassword) {
      id
    }
  }
`

export default compose(
  graphql(UPDATE_PASSWORD_MUTATION, { name: 'updatePasswordMutation' }),
)(ChangePassword)
