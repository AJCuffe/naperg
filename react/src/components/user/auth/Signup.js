import React, { Component } from 'react'
import { AUTH_TOKEN } from '../../../constants/constants'
import { graphql, compose } from 'react-apollo'
import gql from 'graphql-tag'
import SnackBarCustom from '../../nav/SnackBarCustom'
import Password from './Password'
import Paper from '@material-ui/core/Paper'
import Button from '@material-ui/core/Button'
import Input from '@material-ui/core/Input'
import InputAdornment from '@material-ui/core/InputAdornment'
import Icon from '@material-ui/core/Icon'
import LinearProgress from '@material-ui/core/LinearProgress'
import InputLabel from '@material-ui/core/InputLabel'
import FormControl from '@material-ui/core/FormControl'
import { withApollo } from 'react-apollo'

var validator = require('email-validator')


class Signup extends Component {
  state = {
    email: '',
    emailValidation: true,
    inputValidation2: true,
    password: '',
    name: '',
    nameFile: '',
    activeStep: 0,
    maxStep: 3,
    // isPasswordLongEnough: true,
    // hasLowerCase: true,
    // isPasswordActiveStep: false,
    // passwordMinimumLength: 10
  }
  onChange2(e){

    this.setState({
      password: e.target.value,
      // hasLowerCase: this.hasLowerCase(e.target.value),
      // isPasswordLongEnough: this.isPasswordLongEnough(e.target.value)
    })
    // if(this.state.hasLowerCase && this.state.isPasswordLongEnough) {
    //   this.setState({inputValidation2: true})
    // } else {
    //   this.setState({inputValidation2: false})
    // }
  }

  // hasLowerCase(str) {
  //    if(str.toUpperCase() !== str) {
  //      return true
  //    }
  //    return false
  // }

  // isPasswordLongEnough(password) {
  //   if(password.length > this.state.passwordMinimumLength) {
  //     this.setState({isPasswordLongEnough: true})
  //     return true
  //   }
  //   this.setState({isPasswordLongEnough: false})
  //   return false
  // }
  onChange1(e){
    this.setState({ email: e.target.value })
    if (this.validateEmail(e.target.value )) {
      this.setState({emailValidation: true})
    } else {
      this.setState({emailValidation: false})
    }
  }
  validateEmail(email) {
      return validator.validate(email)
      // var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
      // return re.test(String(email).toLowerCase())
  }
  calculateBuffer() {
    let data = ''
    if(this.state.activeStep === 0) {
      data = this.state.name
    }
    if(this.state.activeStep === 1) {
      data = this.state.email
    }
    if(this.state.activeStep === 2) {
      data = this.state.password
    }
    let maxValue = data.length/10 > 1 ? 1 : data.length/10
    return (this.state.activeStep + maxValue) * 100 / this.state.maxStep
  }
  handleNext = () => {
    if(this.state.name) {
      if(this.state.activeStep === 0 ) {
        this.setState({
          activeStep: this.state.activeStep + 1,
        }, () => {
           this.input1.focus()
        })
      }
      if(this.state.activeStep === 1 ) {
        if(this.state.emailValidation ) {
          this.setState({
            activeStep: this.state.activeStep + 1,
          }, () => {
            // this.input2.focus()
            this.setState({isPasswordActiveStep:true})
          })
        }
      }
      if(this.state.activeStep === 2 ) {
        this.setState({
          activeStep: this.state.activeStep + 1,
        }, () => {
          this._confirm()
        })
      }
    }
  };

  handleKey = (data) => {
    if(data.charCode === 13) { //keyPress enter
      this.handleNext()
    }
  }


  render() {
    return (
      <div className='paperOut'>
        <Paper className='paperIn'>
        <h4 className='mv3'>
          Sign Up
        </h4>
        <div className='flex flex-column'>

        <LinearProgress variant='buffer'
          value={this.state.activeStep * 100 / this.state.maxStep }
          valueBuffer={this.calculateBuffer()}
             />
          <br/>
            <div className='tac'>
              <FormControl className={'wrapperAnimate ' + (this.state.activeStep === 0 ? 'focusField' : 'notFocusField')}>
                <InputLabel htmlFor='name'>Your name</InputLabel>
                <Input
                  id='name'
                  value={this.state.name}
                  inputRef={node => this.input0 = node}
                  onChange={e => this.setState({ name: e.target.value })}
                  type='text'
                  onKeyPress={this.handleKey}
                  endAdornment={
                    <InputAdornment position='end'>
                      {this.state.activeStep === 0 && (
                        <Button onClick={this.handleNext} variant='fab' color='primary' mini>
                          <Icon>navigate_next</Icon>
                        </Button>
                      )}
                    </InputAdornment>
                  }
                />
            </FormControl>
            <br/><br/>
            {this.state.activeStep >= 1 && (
              <FormControl className={'wrapperAnimate ' + (this.state.activeStep === 1 ? 'focusField' : 'notFocusField')}>
                <InputLabel htmlFor='email'>Your email address</InputLabel>
              <Input
                id='email'
                value={this.state.email}
                error={!this.state.emailValidation}
                onChange={this.onChange1.bind(this)}
                type='text'
                inputRef={node => this.input1 = node}
                onKeyPress={this.handleKey}
                endAdornment={
                  <InputAdornment position='end'>
                    {this.state.activeStep === 1 && (
                      <Button onClick={this.handleNext} variant='fab' color='primary' mini>
                        <Icon>navigate_next</Icon>
                      </Button>
                    )}
                  </InputAdornment>
                }
                />
              </FormControl>
            )}
            <br/><br/>
            {this.state.activeStep >= 2 && (
              <Password
                handleNext={this.handleNext.bind(this)}
                activeStep={this.state.isPasswordActiveStep}
                onChange2={this.onChange2.bind(this)}/>
            )}
          </div>
        </div>


        <SnackBarCustom ref={instance => { this.child = instance }}/>
      </Paper>
      </div>
    )
  }

  _confirm = async () => {
    const { name, email, password, nameFile } = this.state
      await this.props.signupMutation({
        variables: {
          name,
          email,
          password,
          nameFile,
        },
      })
      .then((result) => {
        const { token, user } = result.data.signup
        this._saveUserData(token, user)

      })
      .catch((e) => {
        this.child._openSnackBar(e.graphQLErrors[0].message)
      })

  }

  _saveUserData = (token, user) => {
    localStorage.setItem(AUTH_TOKEN, token)
    localStorage.setItem('userToken', JSON.stringify(user))
    this.props.client.resetStore().then(data=> {
      this.props.history.push(`/`)
    })
  }
}

const SIGNUP_MUTATION = gql`
  mutation SignupMutation($email: String!, $password: String!, $name: String!, $nameFile: String!) {
    signup(email: $email, password: $password, name: $name, nameFile:$nameFile) {
      token
      user {
        name
        emailvalidated
        id
      }
    }
  }
`


export default compose(
  withApollo,
  graphql(SIGNUP_MUTATION, { name: 'signupMutation' }),

)(Signup)
