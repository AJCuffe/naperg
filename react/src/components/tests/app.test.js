import { configure, shallow, mount, render } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import React from 'react'
import { ApolloProvider } from 'react-apollo'
import Signup from '../user/auth/Signup'
import { Login } from '../user/auth/Login'

import { graphql, compose } from 'react-apollo'
import { withApollo } from 'react-apollo'



configure({ adapter: new Adapter() })


it('Signup', () => {
  const wrapper = shallow(
    <Signup/>
  )
  expect(wrapper).toMatchSnapshot()
})

it('Login', () => {
  const wrapper = shallow(
    <Login/>
  )
  expect(wrapper).toMatchSnapshot()
})

it('renders a email input', () => {
  const wrapper = shallow(<Login />);
  expect(wrapper.find('#email').length).toEqual(1)
})

it('email input', () => {
  const wrapper = shallow(<Login />)
  wrapper.find('#email').simulate('change', {target: {name: 'email', value: 'blah@gmail.com'}});
  expect(wrapper.state('email')).toEqual('blah@gmail.com');
})


describe('Password input', () => {

 it('should respond to change event and change the state of the Login Component', () => {

  const wrapper = shallow(<Login />);
  wrapper.find('#password').simulate('change', {target: {name: 'password', value: 'cats'}});

  expect(wrapper.state('password')).toEqual('cats');
 })
})
//
// import React from 'react';
// import { mount } from 'enzyme';
// import Signup from './user/auth/Signup'
//
// it('Renders the SimpleInput', () => {
//
//   const component = mount(
//     <Signup
//
//     />
//   );
//
//   expect(component).toMatchSnapshot();
// });
