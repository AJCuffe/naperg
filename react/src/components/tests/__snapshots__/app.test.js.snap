// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withApollo(Apollo(Login)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Signup 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withApollo(Apollo(Signup)) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
